### DISCUSSION TITLE
Display a warning about missed backups

### DISCUSSION THREAD

**Discussion by fantom-x** [borgbase/vorta#807] [Ideas]
I do understand that when a computer is not on, there is no backup. But I have backups scheduled every three hours and I happened to not be using the computer at the time of scheduled backups for several days in a row.
There are log messages about that, but it would be beneficial to the users to have a clearly visible warning about missed backups so that the actions can be taken before it is too late.
The pop-up warning could also have a button to perform an immediate backup.

**Comment by m3nu** [comment]
We may merge a pull request that offers an immediate backup when too many were missed. Let me know when you got something to review.

**Comment by fantom-x** [comment]
That would be a Cadillac solution. Vorta does log a message in the logs: if it could display a pop up at the same time, that would mostly solve the problem. 
Do not you think that a reliable backup system needs to be able to warn about missed backups?

**Comment by TheReal1604** [comment]
@fantom-x If I understand you correctly there is also no "rescheduling" in place? Maybe this could also be an easy solution? If the last scheduled was missed, plan an immediate backup after the next wakeup.

**Comment by hutson** [comment]
> But I have backups scheduled every three hours and I happened to not be using the computer at the time of scheduled backups for several days in a row.

This has happened to me a few times as well, scheduled backups outside of the few hours I've had my laptop opened and connected to the internet.

A notification, or other prominent alert, that backups have not executed in `X` days would be a great first step to alleviate my own personal concerns.

**Comment by deermo** [comment]
As a workaround, it is possible to use a service such as healthchecks.io and the post-backup-command from the Vorta GUI. In the case of healthchecks.io, you can even integrate if the backup fails, i.e. returncode > 0
```
curl -m 10 --retry 5 "https://hc-ping.com/YOUR-ID$([ $returncode -ne 0 ] && echo -n /fail)"
```

And if you want to track how long backups take, use this pre-backup-command:
```
curl -m 10 --retry 5 https://hc-ping.com/YOUR-ID/start
```
(I have no relationships to healthchecks.io, it's just what I started using, and other providers might work as well)
